File: backend/src/AMS.Domain/Entities/Assignment.cs
using AMS.Domain.Common;
using AMS.Domain.Enums;

namespace AMS.Domain.Entities;

public class Assignment : BaseEntity
{
    public string Title { get; set; } =  string.Empty;
   
    
    
    public string? Description { get; set; } = string.Empty;
    
    public DateTime Deadline { get; set; }
    
    public int MaximumMarks { get; set; }
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public AssignmentStatus Status { get; set; }

    public Guid TeacherId { get; set; }

    public Guid ClassId { get; set; }

    public Guid SubjectId { get; set; }
    
    
}

File: frontend/src/features/admin/assignments/types/assignment.ts
export interface Assignment {
  id: string;
  title: string;
  teacherName: string;
  teacherId: string;
  classId: string;
  className: string;
  subjectId: string;
  subjectName: string;
  status: number;
  deadline: string;
}

export interface AssignmentDetails {
  id: string;
  title: string;
  description: string | null;
  deadline: string;
  maximumMarks: number;
  status: string;
  createdAt: string;

  teacherId: string;
  teacherName: string;

  classId: string;
  className: string;
  section: string;

  subjectId: string;
  subjectName: string;
}
